walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,151 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.lang.Object;
+import java.util.TreeMap;
+import java.util.Set;
+import java.util.Map;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    //The TreeMap stores pairs of key and value. It sorts pairs in ascending order based on their key.
+    //Key is start time of event and value is event. 
+    //So Events will be sorted based on their start time when they get added to the TreeMap.   
+    TreeMap<Integer, Event> eventsList = new TreeMap<Integer, Event>();
+    
+    //Go through events attendees and request attendees 
+    //and add only the important events to TreeMap.
+    //Important events are events that contain any of the attendees in the request attendees. 
+
+    Iterator i = events.iterator();
+    Iterator iter = request.getAttendees().iterator();
```

Comment:
I'd name these `eventsIter` an `requestIter` so that it's easier to distinguish which iter is which in the for loops